services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/external_dependencies.py

```diff
@@ -0,0 +1,20 @@
+from fastapi import FastAPI
+from servicelib.utils import logged_gather
+
+from .postgres import wait_for_postgres_liveness
+from .rabbitmq import wait_for_rabbitmq_liveness
+from .storage import wait_for_storage_liveness
+
+
+def setup_check_dependencies(app: FastAPI) -> None:
+    async def on_startup() -> None:
+        await logged_gather(
+            *[
+                wait_for_storage_liveness(app),
```

Comment:
I still think that it would be better to have these in a setup for each service.  i.e. `setup_storage` in `setup_rabbit`, etc

Reply:
Yeah this one is not obvious why it's like this.
While I agree with you, that it's better to have the checks delegated to their own modules, to achieve a faster start, we run all of them in parallel for the `dy-sidecar`.

Reply:
I don't think it is useful for faster start (the dy-sidecar already takes less than 7 seconds to start).
Maybe in case of multiple dependencies missing... but even there... not sure that makes sense.